src/hooks/useConnectedDelegate.ts

```diff
@@ -21,6 +22,11 @@ const useConnectedDelegate = () => {
   const [balance, setBalance] = useState<bigint | null>(null);
   const [retries, setRetries] = useState<number>(0);
   const [lastVotingPower, setLastVotingPower] = useState<string | null>(null);
+  useQuery({
```

Comment:
I'm generally conservative when it comes to adding new library dependencies, but in this case, replacing `useEffect` with `useQuery` makes a lot of sense.